src/libre/ast_rewrite.c

```diff
@@ -317,6 +317,30 @@ rewrite_alt(struct ast_expr *n, enum re_flags flags)
 		return 1;
 	}
 
+	if (n->u.alt.count == 2) {
```

Comment:
@hvdijk points out that this would also apply for n > 2, which I hadn't thought of.